src/test/java/com/mongodb/jdbc/MongoDriverTest.java

```diff
@@ -87,113 +90,112 @@ void testDBURL() throws SQLException {
         assertNotNull(d.getUnvalidatedConnection(authDBURL, p));
     }
 
+    private void missingConnectionSettings(
+            MongoDriver d, String connectionUrl, Properties properties) {
+        // Should not return null or throw, even with null properties.
+        try {
+            d.getUnvalidatedConnection(connectionUrl, properties);
+            fail("The connection should fail because a mandatory connection setting is missing.");
+        } catch (SQLException e) {
+            // Expected failure
+            assertEquals(
+                    MongoDriver.CONNECTION_ERROR_SQLSTATE,
+                    e.getSQLState(),
+                    "Expect SQL state "
+                            + MongoDriver.CONNECTION_ERROR_SQLSTATE
+                            + " but got "
+                            + e.getMessage());
```

Comment:
```suggestion
                            + e.getSQLState());
```
Suggestion to match the type, SQL state.

Reply:
You are right!